Makefile

```
@@ -1,7 +1,42 @@
-test:
-	go test -race .
-cover:
-	go test . -coverprofile=cover.out
-	go tool cover -html=cover.out
+#Directory to put `go install`ed binaries in.
+export GOBIN ?= $(shell pwd)/bin
+
+GO_FILES := $(shell \
+	find . '(' -path '*/.*' -o -path './vendor' ')' -prune \
+	-o -name '*.go' -print | cut -b3-)
+
+.PHONY: bench
 bench:
-	go test -bench=.
+	go test -bench=. ./...
+
+.PHONY: build
+build:
+	go build ./...
+
+.PHONY: cover
+cover:
+	go test -coverprofile=cover.out -coverpkg=./... -v ./...
+	go tool cover -html=cover.out -o cover.html
+
+.PHONY: gofmt
+gofmt:
+	$(eval FMT_LOG := $(shell mktemp -t gofmt.XXXXX))
+	@gofmt -e -s -l $(GO_FILES) > $(FMT_LOG) || true
+	@[ ! -s "$(FMT_LOG)" ] || (echo "gofmt failed:" | cat - $(FMT_LOG) && false)
+
+.PHONY: golint
+golint:
+	@cd tools && go install golang.org/x/lint/golint
+	@$(GOBIN)/golint ./...
```

consider breaking installation and execution into separate steps so that `make lint` doesn't always try to install golint.

```suggestion
bin/golint: tools/go.mod
	cd tools && go install golang.org/x/lint/golint

.PHONY: golint
golint: bin/golint
	bin/golint ./...
```

(golint binary has a dependency on tools/go.mod because that defines which version of golint we're using.)

---

Sure, make sense - didn't do it because I thought https://github.com/uber-go/multierr/blob/master/Makefile#L33 wasn't doing it for some smart reason :)

---

No, just oversight